src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsWorkItemEmbededLinkEnricher.cs

```diff
@@ -0,0 +1,145 @@
+            {
+                TfsTeamService teamService = Engine.Target.GetService<TfsTeamService>();
+                TfsConnection connection = (TfsConnection)Engine.Target.InternalCollection;
+
+                var identityService = Engine.Target.GetService<IIdentityManagementService>();
+                var tfi = identityService.ReadIdentity(IdentitySearchFactor.General, "Project Collection Valid Users", MembershipQuery.Expanded, ReadIdentityOptions.None);
+                return identityService.ReadIdentities(tfi.Members, MembershipQuery.None, ReadIdentityOptions.None).ToList();
+            });
+        }
+
+        [Obsolete]
+        public override void Configure(bool save = false, bool filterWorkItemsThatAlreadyExistInTarget = true)
+        {
+            throw new NotImplementedException();
+        }
+
+        [Obsolete]
+        public override int Enrich(WorkItemData sourceWorkItem, WorkItemData targetWorkItem)
+        {
+            string oldTfsurl = Engine.Source.Config.AsTeamProjectConfig().Collection.ToString();
+            string newTfsurl = Engine.Target.Config.AsTeamProjectConfig().Collection.ToString();
+
+            var logTypeName = nameof(TfsWorkItemEmbededLinkEnricher);
+            Log.LogInformation($"{logTypeName}: Fixing embedded mention links on target work item {targetWorkItem.Id} from {oldTfsurl} to {newTfsurl}");
+
+            foreach (Field field in targetWorkItem.ToWorkItem().Fields)
+            {
+                if (field.Value == null
+                    || string.IsNullOrWhiteSpace(field.Value.ToString())
+                    || (field.FieldDefinition.FieldType != FieldType.Html && field.FieldDefinition.FieldType != FieldType.History))
+                {
+                    continue;
+                }
+
+                try
+                {
+                    var anchorTagMatches = Regex.Matches((string)field.Value, RegexPatternLinkAnchorTag);
+                    foreach (Match anchorTagMatch in anchorTagMatches)
+                    {
+                        if (!anchorTagMatch.Success) continue;
+
+                        var href = anchorTagMatch.Groups["href"].Value;
+                        var version = anchorTagMatch.Groups["version"].Value;
+                        var value = anchorTagMatch.Groups["value"].Value;
+
+                        if (string.IsNullOrWhiteSpace(href) || string.IsNullOrWhiteSpace(version) || string.IsNullOrWhiteSpace(value))
+                            continue;
+
+                        var workItemLinkMatch = Regex.Match(href, RegexPatternWorkItemUrl);
+                        if (workItemLinkMatch.Success)
+                        {
+                            var workItemId = workItemLinkMatch.Groups["id"].Value;
+                            Log.LogDebug($"{logTypeName}: Source work item {workItemId} mention link traced on field {field.Name} on target work item {targetWorkItem.Id}.");
```

Comment:
All Logging should be in the format `"My message that contains {item} and {item2}", item, itesm2`! Do not use `$"My message that contains {item} and {item2}"` to pass text into the log strings as this disables [Serilog](https://serilog.net/)'s ability to pass that data as telemetry to Application Insights and for log highlighting.

Reply:
Updated..